mlir/test/Dialect/OpenMP/invalid.mlir

```diff
@@ -1658,4 +1658,77 @@ func.func @omp_target_exit_data(%map1: memref<?xi32>) {
   return
 }
 
+// -----
+
+func.func @omp_target_update_data_if(%if_cond : i1) {
+  // expected-error @below {{`if` clause can appear at most once in the expansion of the oilist directive}}
+  omp.target_update_data if(%if_cond : i1) if(%if_cond : i1)
+  return
+}
```

Comment:
We don't need to add checks for cases for which it isn't possible to create an mlir op. Please remove other such checks.

Reply:
These test that we defined the op properly in TableGen, they guard against invalid op definition/change.

> ... which it isn't possible to create an mlir op.

Also, how do you guarantee that 😛?

I don't see harm in adding conservative testing.

Reply:
The op builder wouldn't let you create an op with two nowait or if clause or whatever that isn't specified in the op definition.

By your argument why not test that it doesn't work for other random clauses as well?

Reply:
> By your argument why not test that it doesn't work for other random clauses as well?

That I agree with. Removed the extra tests. Thanks for the discussion.